packages/x-components/src/x-modules/search-box/components/search-input-placeholder.vue

```diff
@@ -0,0 +1,179 @@
+<template>
+  <div v-show="isQueryEmpty" class="x-search-input-placeholder">
+    <component :is="animation" @after-enter="showNextPlaceholder">
+      <span
+        v-if="animatedPlaceholder && isAnimationVisible"
+        :key="animatedPlaceholder"
+        class="x-search-input-placeholder__placeholder"
+        data-test="animated-placeholder"
+      >
+        {{ animatedPlaceholder }}
+      </span>
+      <span
+        v-else-if="$slots.default"
+        class="x-search-input-placeholder__default-placeholder"
+        data-test="default-placeholder"
+      >
+        <!-- @slot Default static placeholder. -->
+        <slot />
+      </span>
+    </component>
+  </div>
+</template>
+
+<script lang="ts">
+  import Vue from 'vue';
+  import { Component, Prop } from 'vue-property-decorator';
+  import { State } from '../../../components/decorators/store.decorators';
+  import { XOn } from '../../../components/decorators/bus.decorators';
+  import { xComponentMixin } from '../../../components/x-component.mixin';
+  import { searchBoxXModule } from '../x-module';
+
+  /**
+   * Renders a placeholder that is displayed on top of the search input and
+   * can be animated when given an array of animated placeholders.
+   *
+   * @public
+   */
+  @Component({
+    mixins: [xComponentMixin(searchBoxXModule)]
+  })
+  export default class SearchInputPlaceholder extends Vue {
+    @State('searchBox', 'query')
+    public query!: string;
+
+    protected get isQueryEmpty(): boolean {
+      return this.query.length === 0;
+    }
+
+    /**
+     * List of placeholders that will be animated.
+     *
+     * @public
+     */
+    @Prop({ default: () => [] })
+    protected animatedPlaceholders!: string[];
```

Comment:
I'm not really sold on the name. The animation is not required in the component, so they might not get animated. Something more accurate could be `iterativePlaceholders`, `placeholderCandidates` or just `placeholderList`.